corehq/apps/hqadmin/tests/test_pillowtop.py

```diff
@@ -52,22 +55,21 @@ def test_with_blacklist_items(self):
         self.assertEqual(expected_pillows, set([c.name for c in configs_back]))
 
     def test_loading_existing_conf_file(self):
+        config = {'include_groups': ['mvp_indicators']}
 
-        expected_action = {'include_groups': ['mvp_indicators']}
-
-        action = get_single_pillow_action('staging')
-        self.assertEqual(action.to_json(), expected_action)
-
-    def test_loading_no_existing_conf_file(self):
-        action = get_single_pillow_action('foo')
-        self.assertIsNone(action)
+        action = get_single_pillow_action(config)
+        self.assertEqual(action.to_json(), config)
 
     def test_india_server_exclusions(self):
         self.pillowtops['fluff'] = [
             'custom.bihar.models.CareBiharFluffPillow',
             'custom.opm.models.OpmUserFluffPillow',
         ]
+        configs = [
+            {'exclude_groups': ['mvp_indicators', 'fluff']},
+            {'include_pillows': ['custom.bihar.models.CareBiharFluffPillow']},
+        ]
 
-        pillows = [c.name for c in get_pillows_for_env('softlayer', self.pillowtops)]
+        pillows = [c.name for c in get_pillows_for_env(configs, self.pillowtops)]
         self.assertNotIn('OpmUserFluffPillow', pillows)
         self.assertIn('CareBiharFluffPillow', pillows)
```

Comment:
True re the pillowtop version of the command. If we used rsync then it
would only push the diffs.

On 22 April 2016 at 08:48, Cory Zue <notifications@github.com> wrote:

> In corehq/apps/hqadmin/tests/test_pillowtop.py
> <https://github.com/dimagi/commcare-hq/pull/11190#discussion_r60697312>:
>
> >
> > -        pillows = [c.name for c in get_pillows_for_env('softlayer', self.pillowtops)]
> > +        pillows = [c.name for c in get_pillows_for_env(configs, self.pillowtops)]
> >          self.assertNotIn('OpmUserFluffPillow', pillows)
> >          self.assertIn('CareBiharFluffPillow', pillows)
>
> I think the command also needs settings.PILLOWTOPS too (but assume we
> could figure that out). Pushing 15 conf files each deploy might be slow
> though would be great to add checking to only do any work if we have to
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/11190/files/4e187254945fb99dc7ff551ede70919e4e676d3a..87f1dabc38f76857975f79f6c9993d8183008deb#r60697312>
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa